### PR TITLE
Remove element's nullability of array_agg function

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Closes #.

## Rationale for this change

I think the nullability of element in array_agg_* function (in general UDAF, those elements whether null or non-null could be handled in function invoke logic itself) doesn't help much about optimization, but add complexity for the function itself. I propose to remove it until there is any use case that shows the nullability of element is really helpful.

The related change is from https://github.com/apache/datafusion/pull/8055, where introduce `nullable` but place it in a incorrect place. Then, #11093 where change the nullable from list to element. And this PR is going to cleanup nullable of element 😫

### CODE REVIEW

File: datafusion/core/tests/sql/aggregates.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ async fn csv_query_array_agg_distinct() -> Result<()> {
         *actual[0].schema(),
         Schema::new(vec![Field::new_list(
             "ARRAY_AGG(DISTINCT aggregate_test_100.c2)",
-            Field::new("item", DataType::UInt32, false),
```
Comment (alamb) [inline datafusion/core/tests/sql/aggregates.rs:39]: array agg is NULL when there are no inputs, I believe after https://github.com/apache/datafusion/pull/11299 so this change makes sense to me

**Review by alamb — COMMENTED** [review]
This change makes sense to me, but I feel like we are undergoing some API churn. It may be good to ensure we won't have to change this again.

For example, is it the case that for *all* aggregates (both existing and potentially user defined ones) would not want to know if their input is nullable (or is there some other way to figure that out?)

Perhaps we could update the documentation somehow to make the design choice about null inputs --> output computation clearer

@eejbyfeldt  I wonder if you have a chance to review as the original author of https://github.com/apache/datafusion/pull/11093 ?

**Comment by jayzhan211** [comment]
I think in theory UDAF could have `input_nullable`, but I'm not sure about what could be the benefit from it. 🤔

**Review by alamb — APPROVED** [review]
Makes sense to me -- thanks @jayzhan211

**Comment by alamb** [comment]
Let's plan to merge this tomorrow to allow anyone else who may wish to review

**Comment by eejbyfeldt** [comment]
I don't think I am really following the motivation here. For me, the reason to have the correct nullability of the field inside the returned list is the same as having correct nullability for any other field.

> I propose to remove it until there is any use case that shows the nullability of element is really helpful.

If this argument hold here. Then why is it not also an argument to remove the concept of nullability throughout datafusion? (Which to me, it seems like it would be very undesired)

**Comment by jayzhan211** [comment]
@eejbyfeldt 
I see. I agree the motivation here is not really a good reason, but it is my tradeoff about how we can move on for aggregate function.

I think the tradeoff here is having *idea* nullability for possible optimization and having *correct* but simple nullability **for now** and able to bring in *idea* nullability with **actual** optimization in the future.

> For me, the reason to have the correct nullability of the field inside the returned list is the same as having correct nullability for any other field

The nullability of field is actually the reason that matters to me. I'm not sure if there is possible improvement for **nullable element** in aggregate function. If there is any optimization like what you show me previously, I would be happy to keep the nullability and find an idea way to bring it to UDAF.

**Comment by alamb** [comment]
I defer to @jayzhan211  with what to do with this PR. I am fine either way (keep the API in case it might be helpful in the future or remove it and we can add it again if it is needed)

**Comment by jayzhan211** [comment]
I would like to merge this, since the builtin function code will eventually be removed, we can add it for UDAF later on

**Comment by jayzhan211** [comment]
Thanks @alamb and @eejbyfeldt for your suggestion

**Pull request merged**